api-tests/plugins/i18n/content-manager/non-localized-fields.test.api.js

```diff
@@ -0,0 +1,360 @@
+      ])
+      .build();
+
+    await modelsUtils.modifyComponent(compo(true));
+
+    strapi = await createStrapiInstance();
+    rq = await createAuthRequest({ strapi });
+  });
+
+  afterAll(async () => {
+    // Delete all locales that have been created
+    await strapi.db.query('plugin::i18n.locale').deleteMany({ code: { $ne: 'en' } });
+
+    await strapi.destroy();
+    await builder.cleanup();
+  });
+
+  describe('Non localized fields', () => {
+    let documentId = '';
+
+    beforeAll(async () => {
+      // Create a document with an entry in every locale with the localized
+      // field filled in. This field can be different across locales
+      const res = await rq({
+        method: 'POST',
+        url: `/content-manager/collection-types/api::category.category`,
+        body: {
+          name: `Test`,
+        },
+      });
+      documentId = res.body.data.documentId;
+
+      for (const locale of allLocaleCodes) {
+        await rq({
+          method: 'PUT',
+          url: `/content-manager/collection-types/api::category.category/${documentId}`,
+          body: {
+            locale,
+            name: `Test ${locale}`,
+          },
+        });
+      }
+
+      const tagRes = await rq({
+        method: 'POST',
+        url: `/content-manager/collection-types/api::tag.tag`,
+        body: {
+          name: `Test tag`,
+        },
+      });
+      data.tags.push(tagRes.body.data);
+
+      for (const locale of allLocaleCodes) {
+        const localeTagRes = await rq({
+          method: 'PUT',
+          url: `/content-manager/collection-types/api::tag.tag/${tagRes.body.data.documentId}`,
+          body: {
+            locale,
+            name: `Test tag ${locale}`,
+          },
+        });
+
+        data.tags.push(localeTagRes.body.data);
+      }
+    });
+
+    describe.each([
+      [
+        {
+          // Test that when you update a non localized field all
+          // other locales are updated accordingly
+          description: 'Modify a top level non localized field',
+          key: 'nonLocalized',
+          action: {
+            update: 'Update Test',
+          },
+        },
+      ],
+      [
+        {
+          // Test that when you change a field in a non localized compo, all
+          // others are updated accordingly
+          description: 'Modify a field within a non localized component',
+          key: 'nonLocalizedCompo',
+          action: {
+            updateAt: [{ key: 'name', value: 'Compo Name' }],
+          },
+        },
+      ],
+      [
+        {
+          // Test when you change a relation to a localized CT within a non localized component, other
+          // locales should not have the equivalent relation associated to them
+          description: 'Connect a relation to a non localized component',
+          key: 'nonLocalizedCompo',
+          action: {
+            connectRelationAt: 'tag',
+          },
+        },
+      ],
```

Comment:
@Marc-Roig I wanted to check this test case with you. Is this correct? 

- When I connect a relation within a non localized component should we update all other locales with the locale equivalent of the relation if it exists?
- And what if that relation content type has i18n disabled?

In both cases relations are currently not copied across all locales, only top level fields within the non localized component

Reply:
>When I connect a relation within a non localized component should we update all other locales with the locale equivalent of the relation if it exists?

yes! I imagine after connecting the relation, your logic should populate the relation inside the compo, and copy it over the other locales.

> And what if that relation content type has i18n disabled?

that should be handled by the `transformParamsDocumentId` or (transformData) , and will keep the relations as they are